src/WorldStorage/WSSAnvil.cpp

```diff
@@ -398,9 +401,21 @@ bool cWSSAnvil::LoadChunkFromNBT(const cChunkCoords & a_Chunk, const cParsedNBT
 			MetaData = GetSectionData(a_NBT, Child, "Data", ChunkBlockData::SectionMetaCount),
 			BlockLightData = GetSectionData(a_NBT, Child, "BlockLight", ChunkLightData::SectionLightCount),
 			SkyLightData = GetSectionData(a_NBT, Child, "SkyLight", ChunkLightData::SectionLightCount);
-		if ((BlockData != nullptr) && (MetaData != nullptr) && (SkyLightData != nullptr) && (BlockLightData != nullptr))
+		if ((BlockData != nullptr) && (SkyLightData != nullptr) && (BlockLightData != nullptr))
 		{
-			Data.BlockData.SetSection(*reinterpret_cast<const ChunkBlockData::SectionType *>(BlockData), *reinterpret_cast<const ChunkBlockData::SectionMetaType *>(MetaData), static_cast<size_t>(Y));
+			BlockState Blocks[ChunkBlockData::SectionBlockCount];
+			for (size_t I = 0; I < ChunkBlockData::SectionBlockCount; I++)
+			{
+				unsigned char Meta = 0;
+				if (MetaData != nullptr)
+				{
+					Meta = static_cast<unsigned char>(MetaData[I]);
```

Comment:
Don't forget to say `cChunkDef::ExpandNibble(MetaData, i)`

The meta values in the old format at packed two entries into one byte.

Reply:
There we go